actions/run.py

```diff
@@ -24,17 +24,25 @@ def _get_request(self, params):
             headers=headers, data=params
         )
 
-        results = response.json()
-        if results['result_code'] is not 1:
+        self.logger.debug("status: %s body: %s" % (
+            response.status_code, response.text))
+
+        if response.status_code is not 200:
             failure_reason = (
                 'Failed to perform action. status code: %s; response body: %s' % (
                     response.status_code,
-                    response.json()
+                    response.text
                 )
             )
             self.logger.exception(failure_reason)
             raise Exception(failure_reason)
 
+        results = response.json()
+        if results['result_code'] is not 1:
```

Comment:
Overall, the change itself looks controversial.

From the one side, to conform with RESTful API principles, `200 OK` must not be used to communicate errors in response body. But AC does this, and we need a workaround like one you provided.

From the other side, while this fixes specific issue you're mentioning, I'm afraid this could break some other cases with AC we don't know about yet when error is more fatal.

Reply:
We just log the error though and don't actually change the result of the action so this should be fine, right?

Reply:
Edit: Yeah, it looks like it - https://www.activecampaign.com/blog/detecting-api-errors/

Reply:
Throwing the error is a big problem for a client: there is no `return` ob
Specifically in serverless case, exception leads to Lambda unconditional "Failed" state;  that stops StepFunction can't handle the results. Even in the context of StackStorm/Mistral my example with `contact_add` stands: if I want to handle it as a warning 

I like the idea of tuple return `(False, result)` I didn't do it at first as it has a higher chance of breaking things that work as return object structure changes. 

Let me redo it with that `(False, result)` change. And bump up the version. Alternatively I switch to using contact_change action to not delay blog.